lib/kubeclient.rb

```diff
@@ -67,13 +69,26 @@ def all_entities
     private
 
     def validate_auth_options(opts)
+      bearer_token_keys = [:bearer_token, :bearer_token_file]
+      exclusive_keys = bearer_token_keys << :user
+
+      return if exclusive_keys.none? { |s| opts.key?(s) }
+
+      msg = 'Invalid auth options: specify only one of user/password,' \
+            ' bearer_token or bearer_token_file'
+      fail ArgumentError, msg unless exclusive_keys.one? { |s| opts.key?(s)  }
+
+      msg = 'Basic auth requires both user & password'
+      fail ArgumentError, msg if opts.key?(:user) && !opts.key?(:password)
+    end
+
+    def validate_bearer_token_file(bearer_token_file)
```

Comment:
This happened during refactoring due to line length - should be ok to put on same line again now. Will do it